### PR TITLE
Add isDelete, isPatch check , see #296

### PR DESCRIPTION
**Pull request by netroby**

### CODE REVIEW

**Comment by netroby** [comment]
@laruence , may we add these two method? have a look?

**Comment by laruence** [comment]
should be fine, but I am not sure is there any real usage.

**Comment by netroby** [comment]
We implement API in our company. support full RESTful style requests. so we implement using php code wrapper yaf. but i   wish these methods could be add upstream .to gain better performance. Best wishes. happy new year.

**Comment by netroby** [comment]
Here some reference 
http://www.restapitutorial.com/lessons/httpmethods.html

**Comment by netroby** [comment]
@laruence can these method merged with yaf official repository?

**Comment by netroby** [comment]
@laruence . can this PR be merged? there no any BC break

**Comment by laruence** [comment]
is that possible to make some phpts?  using builtin webserver, you could find examples in php-src/sap/cli/tests/, thanks

**Comment by netroby** [comment]
@laruence I had add three tests. and run passed on my machine.
```
PASS Bug #63381 ($_SERVER['SCRIPT_NAME'] changed by yaf) [tests/bug63381.phpt] 
PASS Bug #63438 (Strange behavior with nested rendering) [tests/bug63438.phpt] 
PASS Bug #63900 (Segfault if separated action executes failed) [tests/bug63900.phpt] 
PASS Bug #70913 (Segfault while new Yaf_Controller) [tests/bug70913.phpt] 
PASS Check isDelete [tests/http_is_delete.phpt] 
PASS Check isGet [tests/http_is_get.phpt] 
PASS Check isPatch [tests/http_is_patch.phpt] 
PASS ISSUE #134 (Segfault while calling assemble) [tests/issue134.phpt] 
PASS Issue #163 (forward from init controller) [tests/issue163.phpt] 
SKIP ISSUE #231 (php-fpm worker core dump BUG) [tests/issue231.phpt] reason: CGI not available
SKIP ISSUE #232 (Segfault with Yaf_Route_Simple) [tests/issue232.phpt] reason: CGI not available
PASS ISSUE #297 (Yaf_Loader fail to load namespace class name) [tests/issue297.phpt] 
PASS ISSUE #303 (local variable is overrided by view renderring) [tests/issue303.phpt] 
=====================================================================
TIME END 2017-03-02 02:02:21

```

**Comment by netroby** [comment]
And test php version with 
```
php --version
PHP 7.0.15-0ubuntu0.16.04.2 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.15-0ubuntu0.16.04.2, Copyright (c) 1999-2017, by Zend Technologies

```

Step: 
1. build php
2. build this yaf + pr 
3. install yaf.so
4. make test

File: tests/php.ini [hunk 1]
```diff
@@ -0,0 +1 @@
+extension=yaf.so
```
Comment (laruence) [inline tests/php.ini]: this should not be included

**Comment by laruence** [comment]
nevermind, I will take care of this

**Comment by netroby** [comment]
Thanks. While running make test. if not provided -c php.ini for php built-in server , i can not load yaf.so extension. so i add this. by add this. the php built-in server will load yaf, then the test will running normal

**Pull request merged**